docs/pages/api-reference/operators/window.md

```diff
@@ -0,0 +1,60 @@
+---
+title: Window
+order: 0
+status: published
+---
+### Window
+
+<Divider>
+<LeftSection>
+Operator to do aggregation of events into windows based on their timestamps. Window operator must always be preceded by a [groupby](/api-reference/operators#groupby) operator.
+
+#### Parameters
+
+<Expandable title="type" type="Window">
+
+The type of window to group event on. Possible values are `"hopping"`, `"tumbling"`, `"session"`.
+
+`"hopping"`: Take in two parameters `"duration"` and `"stride"`, the `"duration"` parameters indicate how long the window is while `"gap"` indicate how far apart the window should be. Hopping window can overlap meaning one event can belong to two different windows. `"stride"` should be smaller than `"duration"`.
+
+`"tumbling"`: Take in one parameter `"duration"` indicate how long the window is. The tumbling windows do not overlapped and cover the whole range. 
+
+`"session`: Take in one parameter `"gap"` indicating how close together two events are to be group into one window. Session window do not overlapped and always at least `"gap"` apart.
```

Comment:
Session windows are dynamically sized based on event patterns, using the gap parameter.

gap: Determines the timeout for inactivity that defines the boundary of a window. Events that occur closer together than the specified gap are grouped into the same window. This approach allows windows to be created based on the actual occurrence of events, with windows being at least gap duration apart.